internal/crypto/requests.go

```diff
@@ -14,5 +18,34 @@ func VerifyRequestSignatures[B neofscrypto.ProtoMessage](req neofscrypto.SignedR
 		st.SetMessage(err.Error())
 		return st
 	}
+
 	return nil
 }
+
+// AuthenticateRequest checks whether all request signatures are set and valid.
+// Returns request author ID along with public key if so, or
+// [apistatus.SignatureVerification] error otherwise.
+func AuthenticateRequest[B neofscrypto.ProtoMessage](req neofscrypto.SignedRequest[B]) (user.ID, []byte, error) {
+	if err := VerifyRequestSignatures(req); err != nil {
+		return user.ID{}, nil, err
+	}
+
+	var sig *refs.Signature
```

Comment:
This part can just be a separate method called from `acl/v2` for a request. It doesn't reuse much of existing code anyway, the logic is pretty separate from `VerifyRequestSignatures`.